app/preprints/-components/submit/author-assertions/public-preregistration/component.ts

```diff
@@ -0,0 +1,109 @@
+        } as PreregistationLinkInfoOption,
+        {
+            key: 'prereg_designs',
+            value: this.intl.t('preprints.submit.step-three.public-preregistration-link-info-designs'),
+        } as PreregistationLinkInfoOption,
+        {
+            key: 'prereg_analysis',
+            value: this.intl.t('preprints.submit.step-three.public-preregistration-link-info-analysis'),
+        } as PreregistationLinkInfoOption,
+        {
+            key: 'prereg_both',
+            value: this.intl.t('preprints.submit.step-three.public-preregistration-link-info-both'),
+        } as PreregistationLinkInfoOption,
+    ];
+
+
+    publicPreregistrationOptions = [
+        {
+            inputValue: PreprintPreregLinksEnum.YES,
+            displayText: this.intl.t('general.available'),
+        } as RadioButtonOption,
+        {
+            inputValue: PreprintPreregLinksEnum.NO,
+            displayText: this.intl.t('general.no'),
+        } as RadioButtonOption,
+        {
+            inputValue: PreprintPreregLinksEnum.NOT_APPLICABLE,
+            displayText: this.intl.t('general.not-applicable'),
+        } as RadioButtonOption,
+    ];
+
+    public get displayPublicPreregistrationWhyNoStatement(): boolean {
+        return this.args.changeSet.get('hasPreregLinks') === undefined ?
+            false :
+            !this.displayPublicPreregistrationLinks;
+    }
+
+    public get displayPublicPreregistrationLinks(): boolean {
+        return this.args.changeSet.get('hasPreregLinks') === undefined ?
+            false :
+            this.args.changeSet.get('hasPreregLinks')  === PreprintPreregLinksEnum.YES;
+    }
+
+    @action
+    public updatePublicPreregistrationLinks(links: string[]): void {
+        this.args.changeSet.set('preregLinks', links);
+        this.args.validate();
+    }
+
+    @action
+    public updatePublicPreregistrationOptions(): void {
+        if (this.args.changeSet.get('hasPreregLinks') === PreprintPreregLinksEnum.YES) {
+            this.args.changeSet.set('whyNoPrereg', '');
+            this.isPublicPreregistrationWhyNoStatementDisabled = false;
+        } else if (this.args.changeSet.get('hasPreregLinks') === PreprintPreregLinksEnum.NO) {
+            this.args.changeSet.set('whyNoPrereg', '');
+            this.isPublicPreregistrationWhyNoStatementDisabled = false;
+            this.placeholder = this.intl.t('preprints.submit.step-three.public-preregistration-no-placeholder');
+        } else {
+            this.isPublicPreregistrationWhyNoStatementDisabled = true;
+            this.args.changeSet.set('whyNoPrereg',
+                this.intl.t('preprints.submit.step-three.public-preregistration-na-placeholder'));
+            this.placeholder = this.intl.t('preprints.submit.step-three.public-preregistration-na-placeholder');
+        }
+
+        this.args.validate();
+    }
+
+    @action
+    public updatePreregistrationLinkInfo(linkInfo: PreregistationLinkInfoOption ): void {
```

Comment:
Is this `linkInfo` type correct? It looks like it should be `"prereg_designs" | "prereg_analysis" | "prereg_both"` (could be another use case for a `PreregLinkInfoEnum`)